File: diffbot_ws/src/diffbot_localization/diffbot_localization/imu_republisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
import time
from sensor_msgs.msg import Imu

imu_pub_ = None

def imuCallback(imu_msg):
    global imu_pub_
    imu_msg.header.frame_id = "base_link_ekf"
    imu_pub_.publish(imu_msg)

def main():
    global imu_pub_
    rclpy.init()
    node = Node("imu_republisher_node")
    time.sleep(1)

    node.declare_parameter("imu_topic", "imu/out")  # default = imu/out (simulation)
    imu_topic = node.get_parameter("imu_topic").get_parameter_value().string_value

    imu_pub_ = node.create_publisher(Imu, "imu_ekf", 10)
    imu_sub_ = node.create_subscription(Imu, imu_topic, imuCallback, 10)

    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: diffbot_ws/src/diffbot_localization/src/imu_republisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <sensor_msgs/msg/imu.hpp>

using namespace std::chrono_literals;

rclcpp::Publisher<sensor_msgs::msg::Imu>::SharedPtr imu_pub_;

void imuCallback(const sensor_msgs::msg::Imu & imu_msg)
{
    sensor_msgs::msg::Imu new_imu_msg;
    new_imu_msg = imu_msg;
    new_imu_msg.header.frame_id = "base_link_ekf";
    imu_pub_->publish(new_imu_msg);
}

int main(int argc, char * argv[])
{
    rclcpp::init(argc,argv);
    auto node = rclcpp::Node::make_shared("imu_republisher_node");
    rclcpp::sleep_for(1s);

    imu_pub_ = node->create_publisher<sensor_msgs::msg::Imu>("imu_ekf",10);
    auto imu_sub_ = node->create_subscription<sensor_msgs::msg::Imu>("imu_data",10, imuCallback);

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}